scripts/lib/CIME/case_st_archive.py

```diff
@@ -192,10 +207,13 @@ def _archive_history_files(case, archive, archive_entry,
     rundir = case.get_value("RUNDIR")
     for suffix in archive.get_hist_file_extensions(archive_entry):
         for i in range(ninst):
-            if compname == 'dart':
-                newsuffix = casename + suffix
-            elif compname.find('mpas') == 0:
-                newsuffix = compname + '.*' + suffix
+            if compname.find('mpas') == 0:
+               newsuffix = compname + '.*' + suffix
```

Comment:
While this is not a new problem, I'm wondering if this logic is correct with respect to mpas: It looks to me like this code block cannot handle having an mpas component together with multi-instance?